src/lib/stack.js

```diff
@@ -1,231 +1,70 @@
-  '405': MethodNotAllowedException,
-  '500': ServerErrorException
-}
-
-function getApps() {
-  return fetchJSON(`${COZY_URL}/apps/`, fetchOptions()).then(json => {
-    if (json.error) throw new Error(json.error)
-    else return json.data
-  })
-}
-
-function fetchJSON(url, options) {
-  return fetch(url, options).then(res => {
-    if (typeof errorStatuses[res.status] === 'function') {
-      throw new errorStatuses[res.status]()
-    }
-
-    return res.json()
-  })
-}
-
-// fetch function with the same interface than in cozy-client-js
-function cozyFetchJSON(cozy, method, path, body) {
-  const requestOptions = Object.assign({}, fetchOptions(), {
-    method
-  })
-  requestOptions.headers['Accept'] = 'application/json'
-  if (method !== 'GET' && method !== 'HEAD' && body !== undefined) {
-    if (requestOptions.headers['Content-Type']) {
-      requestOptions.body = body
-    } else {
-      requestOptions.headers['Content-Type'] = 'application/json'
-      requestOptions.body = JSON.stringify(body)
-    }
-  }
-  return fetchJSON(`${COZY_URL}${path}`, requestOptions).then(json => {
-    const responseData = Object.assign({}, json.data)
-    if (responseData.id) responseData._id = responseData.id
-    return Promise.resolve(responseData)
-  })
-}
-
-function getStorageData() {
-  return fetchJSON(`${COZY_URL}/settings/disk-usage`, fetchOptions())
-    .then(json => {
-      return {
-        usage: parseInt(json.data.attributes.used, 10),
-        // TODO Better handling when no quota provided
-        quota: parseInt(json.data.attributes.quota, 10) || 100000000000,
-        isLimited: json.data.attributes.is_limited
-      }
-    })
-    .catch(() => {
-      throw new UnavailableStackException()
-    })
-}
-
-function getContext(cache) {
-  return () => {
-    return cache['context']
-      ? Promise.resolve(cache['context'])
-      : fetchJSON(`${COZY_URL}/settings/context`, fetchOptions())
-          .then(context => {
-            cache['context'] = context
-            return context
-          })
-          .catch(error => {
-            if (error.status && error.status === 404) cache['context'] = {}
-          })
+import internal from 'lib/stack-internal.js'
+import client from "lib/stack-client"
+
+
+/**
+ * Reference to the current client depending
+ * on which one has been initialized
+ *
+ * @private
+ * @TODO We should set it to `undefined`
+ * and throw an error when it is not initialized
+ * Leaving with that default for today
+ * so I do not need to rewrite deeply the tests
+ * (they test some components without initializing
+ * the stack client first, and rely on non-initialized
+ * legacy client behaviour)
+ */
+let stack = internal
+
+/**
+ * Get the current stack client (legacy or cozy-client based)
+ * based on which one has been initialized
+ *
+ * @returns {Object} functions to call the stack
+ */
+const current = function() {
+  if (stack === undefined) {
+    throw new Error("client not initialized in cozy-bar")
```

Comment:
Can we have this case, as `stack` is initialized to `internal`?

Reply:
nope, we don't. 
`stack`should not be set as `internal`by default. I did it to keep compatibility with the legacy tests somewhere else in the code base. I choosed to keep this lines and to put a big TODO warning upper in the file.